libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,144 @@
+                  new UserState(storage ?? new MemoryStorage()))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="rootDialogId">The id of the dialog to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="botFrameworkAuthentication">A <see cref="BotFrameworkAuthentication"/> used to obtain a client for making calls to Bot Builder Skills.</param>
+        /// <param name="skillConversationIdFactoryBase">A <see cref="SkillConversationIdFactoryBase"/> implementation.</param>
+        /// <param name="conversationState">A <see cref="ConversationState"/> implementation.</param>
+        /// <param name="userState">A <see cref="UserState"/> implementation.</param>
+        public AdaptiveDialogBot(
+            ResourceExplorer resourceExplorer,
+            string rootDialogId,
+            string defaultLocale,
+            ILogger<AdaptiveDialogBot> logger,
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            SkillConversationIdFactoryBase skillConversationIdFactoryBase,
+            ConversationState conversationState,
+            UserState userState)
+        {
+            _resourceExplorer = resourceExplorer;
+            _rootDialogId = rootDialogId ?? "Main.dialog";
+            _defaultLocale = defaultLocale ?? DefaultLocale;
+            _logger = logger ?? NullLogger<AdaptiveDialogBot>.Instance;
+            _botFrameworkAuthentication = botFrameworkAuthentication ?? BotFrameworkAuthenticationFactory.Create();
+
+            _skillConversationIdFactoryBase = skillConversationIdFactoryBase;
+            _conversationState = conversationState;
+            _userState = userState;
+        }
+
+        async Task IBot.OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"IBot.OnTurnContext for dialog '{_rootDialogId}'");
+
+            using (var botFrameworkClient = _botFrameworkAuthentication.CreateBotFrameworkClient())
+            {
+                // Set up the TurnState the Dialog is expecting
+                SetUpTurnState(turnContext, botFrameworkClient);
+
+                // Load the Dialog from the ResourceExplorer
+                var rootDialog = await CreateDialogAsync(cancellationToken).ConfigureAwait(false);
+
+                // Run the Dialog
+                await rootDialog.RunAsync(turnContext, turnContext.TurnState.Get<ConversationState>().CreateProperty<DialogState>("DialogState"), cancellationToken).ConfigureAwait(false);
```

Comment:
Kind of tempting to do an overload of RunAsync that drops the second argument. I mean we know ConversationState is on the TurnState so the internals of RunAsync can just find it there. We are assuming a contract on TurnState at this point.